### ISSUE TITLE
Amazon Firestick 4K+ and Chromecast with Google TV

### ISSUE THREAD

**Issue by pwright225** [NineWorlds/serenity-android#446]
unless i am incorrect seems like the newest device this program that we love will work on is the Amazon Firestick 4K (NOT +).

just want to know if and when this will work on Amazon Firestick 4K+ and Chromecast with Google TV?

the program loads but then just spins and seemingly connects to server but then spins forever while attempting to display libraries.  whereas on the 4K (NOT +) it just loads up and works instantly.

i have of course remembered to allow their IP without auth, i have tried both whole subnet and specific IP, which again works either way with the older 4K but not the newer one or the Chromecast.

**Comment by kingargyle** [comment]
@pwright225 if you are using Plex going forward I am dropping support for Plex with version 3.0 (currently under active development) and only supporting Emby Media Server which has similar functionality as plex but a documented and supported API for application and extension developers.

I have a Google TV with Chromecast, that I use daily myself, and know that the latest development version works on that device, as well a the Nvidia Shield, and TCL Android TV televisions.

It should still work on any Fire TV stick, I don't currently have a fire 4K stick but do plan to have it working.

I'm working on a fundamental UI design over the coming holidays, so once I have that done, I'll post a version for testing.

Once I get a Fire 4K stick to test with I can try to work out any issues there.

TLDR; should work on Google TV with Chromecast, untested on Fire 4K stick.

**Comment by kingargyle** [comment]
If using Emby I did have one issue, where the IP address wasn't discoverable.   Had to go into the Emby Server settings, and specifically tell it what IP address to broadcast to clients looking for the server.  This can be done through the web interface in the network section of Emby Server.   Once that was done, Serenity was able to pick up the server.

**Comment by pwright225** [comment]
so does the version you described(latest development) work on plex? and what is the  exact version so i can try to track it down, or can you link it.?

i tried emby way back when, and it wasn't ready yet, and now i am pretty committed to plex. so i don't picture switching anytime soon.

maybe when i can afford a new server with more power, i can run both side by side without issue.

**Comment by kingargyle** [comment]
The main develop branch code, works, but no public consumable APK has been developed from that branch for a while.   The problem with supporting Plex is the fact that they don't really have a client library or a really good documentation around the client portion.  I had to reverse engineer the version that I have that works, and seriously doubt that it works with existing Plex servers that well.

The last development release versions I did, were here:

https://github.com/NineWorlds/serenity-android/wiki/Downloads

Again, the 2.x series is pretty much feature complete and the last Plex supported version I'll do.   I don't personally use Plex any more so have switched everything myself to Emby.

Code can always be forked and is under a MIT license which pretty much lets anyone do what they want with it.

**Comment by pwright225** [comment]
Thanks! but i guess that is that, i just tried the v2.0.0 Milestone 8  and it didn't fix the issue. the end of an era, we really appreciated your app while it lasted.  cheers and great work!

**Comment by kingargyle** [comment]
I'll leave this open so that I know to make sure to test 3.0 on the affected devices.

**Comment by kingargyle** [comment]
Installed the latest version on my Fire TV 4K stick and app works as expected.   Working on Google Chromecast.   Will drop support for the original Amazon Fire TV stick as while it works, the experience is not up to snuff.  The original stick is just to slow.

Known working devices:

* Google Chromecast 4k
* Amazon Fire TV 4k
* Nvidia Shield (original)
* TCL Beyond TV 5 Android TV build in to 4 Series.

Closing this for now for Serenity v3.
